internal/gps/solver.go

```diff
@@ -109,6 +109,10 @@ type SolveParameters struct {
 	// solver will generate informative trace output as it moves through the
 	// solving process.
 	TraceLogger *log.Logger
+
+	// stdLibFn is the function to use to recognize standard library import paths.
+	// Only overridden for tests. Defaults to paths.IsStandardImportPath if nil.
+	stdLibFn func(string) bool
```

Comment:
Ah, cool solution! Having the unexported field on something that's so publicly used it a little ugly, but it's less ugly than any alternatives we came up with, soooo 😄